cachi2/core/package_managers/pip.py

```diff
@@ -1386,14 +1387,21 @@ def _download_dependencies(
     pip_deps_dir = output_dir.join_within_root("deps", "pip")
     pip_deps_dir.path.mkdir(parents=True, exist_ok=True)
 
-    downloads = []
+    downloaded = []
+    to_download: list[DistributionPackageInfo] = []
 
     for req in requirements_file.requirements:
         log.info("Downloading %s", req.download_line)
 
         if req.kind == "pypi":
-            download_info = _download_pypi_package(req, pip_deps_dir, PYPI_URL)
-            _check_metadata_in_sdist(download_info["path"])
+            source, wheels = _process_package_distributions(req, pip_deps_dir, allow_binary)
+            if allow_binary:
+                to_download.extend(wheels)
```

Comment:
We still want to avoid redownloading wheels, right? I understood that we decided to live with the corner case of "the order of input packages might affect hash checking" (e.g. switching the order in which project_a and project_b are processed in [scenario 1](https://github.com/containerbuildsystem/cachi2/pull/255#discussion_r1297342246)).